File: 345-reverse-vowels-of-a-string/reverse-vowels-of-a-string.py
class Solution(object):
    def reverseVowels(self, s):
        """
        :type s: str
        :rtype: str
        """
        word = list(s)
        l = 0
        r = len(s)-1
        vowels = "ueoaiUEOAI"
        while l<r:
            while l<r and vowels.find(word[l]) == -1: 
                l += 1
            while l<r and vowels.find(word[r]) == -1:
                r -= 1
            word[l], word[r] = word[r], word[l]
            l += 1
            r -= 1 
        return "".join(word) 

File: 345-reverse-vowels-of-a-string/reverse-vowels-of-a-string.java
class Solution {
    public String reverseVowels(String s) {
        char[] string = s.toCharArray();
        int l = 0, r = string.length - 1;
        String vowels = "aeiouAEIOU";
        while(l<r){
            while(l<r && vowels.indexOf(string[l]) == -1){
                l++;
            }
            while(l<r && vowels.indexOf(string[r]) == -1){
                r--;
            }
            char tmp = string[l];
            string[l] = string[r];
            string[r] = tmp;

            l++;
            r--;
        }
        String res = new String(string);
        return res;
    }
}